src/org-manager/components/organisation-details-info/organisation-details-info.component.ts

```diff
@@ -8,14 +10,66 @@ import { OrganisationVM} from 'src/org-manager/models/organisation';
   selector: 'app-org-details-info',
   templateUrl: './organisation-details-info.component.html'
 })
-export class OrganisationDetailsInfoComponent {
+export class OrganisationDetailsInfoComponent implements OnInit {
 
   @Input() public org: OrganisationVM;
   @Input() public orgDeletable: boolean;
   @Output() public approveEvent: EventEmitter<OrganisationVM> = new EventEmitter();
   @Output() public deleteEvent: EventEmitter<OrganisationVM> = new EventEmitter();
+  @Output() public reviewEvent: EventEmitter<OrganisationVM> = new EventEmitter();
 
-  constructor() {}
+  public formGroup: FormGroup;
+  public submitted = false;
+  public errorMessage: ErrorMessage;
+  private readonly genericError = 'There is a problem';
+  private readonly radioSelectedControlName = 'radioSelected';
+  public readonly registrationRequest: DisplayedRequest[];
+
+  constructor(private readonly fb: FormBuilder) {
+    this.registrationRequest = [
+      { request: RequestType.APPROVE_REQUEST, checked: false },
+      { request: RequestType.REJECT_REQUEST, checked: false },
+      { request: RequestType.REVIEW_REQUEST, checked: false },
+    ];
+  }
+
+  public ngOnInit(): void {
+    this.formGroup = this.fb.group({
+      radioSelected: new FormControl(null, Validators.required),
+    });
+  }
+
+  public onSubmit(): void {
+    this.submitted = true;
+    if (this.formGroup.get(this.radioSelectedControlName).invalid) {
+      this.errorMessage = {
+        title: this.genericError,
+        description: RequestErrors.NO_SELECTION,
+        fieldId: 'options'
+      };
+    }
+    if (this.formGroup.valid) {
+      const radioSelectedValue = this.formGroup.get(
+        this.radioSelectedControlName
+      ).value;
+
+      switch (radioSelectedValue) {
+        case RequestType.APPROVE_REQUEST:
+          this.approveOrganisation(this.org);
+          break;
+        case RequestType.REJECT_REQUEST:
+          this.deleteOrganisation(this.org);
+          break;
+        default :
+          this.reviewOrganisation(this.org);
+          break;
+      }
+    }
```

Comment:
Maybe to reduce nesting you could have...

```
if (this.formGroup.invalid) {
return;
}
const radioSelectedValue = this.formGroup.get(
  this.radioSelectedControlName
).value;

switch (radioSelectedValue) {
  case RequestType.APPROVE_REQUEST:
    this.approveOrganisation(this.org);
    break;
  case RequestType.REJECT_REQUEST:
    this.deleteOrganisation(this.org);
    break;
  default :
    this.reviewOrganisation(this.org);
    break;
```

Reply:
done